setup.py

```diff
@@ -7,7 +7,7 @@
     long_description = fh.read()
 
 install_requires = [
-    line.split("==")[0] for line in open("requirements.txt", "r") if ".git" not in line
+    line for line in open("requirements.txt", "r") if ".git" not in line
```

Comment:
I really don't like this approach -- it's magic, and makes the production environment differ from dev, potentially leading to hard-to-debug conditions.

Since we've eliminated the repo reference from requirements.txt, I don't think this PR is needed at all.

Two thoughts if you do want to proceed:
1. remove requirements from setup.py entirely, in favor of specifying as much as possible in `pyproject.toml` as proposed in #372 
2. annotating this with a huge, ugly comment, so it's very obvious that magic is occuring.